src/Workspaces/Core/Portable/EmbeddedLanguages/EmbeddedLanguageCommentOptions.cs

```diff
@@ -4,48 +4,35 @@
 
 using System;
 using System.Collections.Generic;
+using System.Collections.Immutable;
 using System.Linq;
 using System.Reflection;
-using System.Text.RegularExpressions;
 
-namespace Microsoft.CodeAnalysis.EmbeddedLanguages.LanguageServices
+namespace Microsoft.CodeAnalysis.EmbeddedLanguages
 {
     /// <summary>
     /// Helps match patterns of the form: language=name,option1,option2,option3
-    /// 
+    /// <para/>
     /// All matching is case insensitive, with spaces allowed between the punctuation. Option values will be or'ed
     /// together to produce final options value.  If an unknown option is encountered, processing will stop with
     /// whatever value has accumulated so far.
-    /// 
+    /// <para/>
     /// Option names are the values from the TOptions enum.
     /// </summary>
-    internal struct LanguageCommentDetector<TOptions> where TOptions : struct, Enum
+    internal static class EmbeddedLanguageCommentOptions<TOptions> where TOptions : struct, Enum
     {
         private static readonly Dictionary<string, TOptions> s_nameToOption =
             typeof(TOptions).GetTypeInfo().DeclaredFields
```

Comment:
is reflection the only way to do this?

Reply:
i can't think of another way.  this is also shipped code.

Reply:
also, this is simply done once and cached per Options type.